app/controllers/likes_controller.rb

```diff
@@ -1,6 +1,14 @@
 class LikesController < ApplicationController
   before_action :set_like, only: %i[ show edit update destroy ]
 
+  before_action :is_an_authprized_user, only: [:destroy, :create]
+
+  def is_an_authprized_user
```

Comment:
The logic within is_an_authprized_user seems inverted. As written, it redirects when the conditions for being authorized are met, which is the opposite of what you typically want. You should redirect if the user is not authorized. Consider reversing the logic to correctly handle unauthorized access.